docs/proposals/admission_control_changes.md

```diff
@@ -0,0 +1,485 @@
+would therefore deprecate the `--admission-control` setting from
+`pkg/genericapiserver/options.ServerRunOptions`. During the deprecation period,
+if this flag is set, we will display a warning while continuing to honor the values
+specified by the flag (i.e., the administrator can override the hard-coded ordering).
+
+
+## One config file per plugin
+
+All admission plugins currently share a single config file, specified via
+`--admission-control-config-file`. This is problematic if multiple plugins
+use the same names for their configuration sections. For
+example, if plugin `Foo` wants
+
+```yaml
+myConfig:
+  someField: red
+```
+
+and plugin `Bar` wants
+
+```yaml
+myConfig:
+  someField:
+  - one
+  - two
+  - three
+```
+
+this won't work because of the naming collision and the type of `someField` is
+different for the two plugins. What we'd like to see instead is:
+
+1. deprecate `--admission-control-config-file`
+1. replace it with `--admission-plugin PLUGIN=CONFIG_FILE` (or an equivalent
+   setup in a structured configuration, once we've fully moved over)
+
+You would specify `--admission-plugin` once per plugin that you want to
+configure. An example of this would be
+
+```
+--admission-plugin AlwaysPullImages=/path/to/config.yaml --admission-plugin MyPlugin=/some/other/config.yaml
+```
+
+An example of a structured configuration for admission control inside an
+apiserver configuration file might look like this:
+
+```yaml
+admissionConfig:
+  AlwaysPullImages:
+    configurationFile: /path/to/config.yaml
+  MyPlugin:
+    configurationFile: /some/other/config.yaml
+  
+```
+
+## Ability to enable/disable individual plugins
+
+With a fixed list of plugins enabled by default, administrators will need to be
+able to override the defaults and disable certain plugins that are normally on,
+and enable plugins that are normally off. We are proposing that the admission
+control section in the apiserver configuration file include support for this. We
+can use `enabled` as seen below:
+
+```yaml
+admissionConfig:
+  AlwaysPullImages:
+    # disabled by default, so enable it
+    enabled: true
+    configurationFile: /path/to/config.yaml
+  MyPlugin:
+    # enabled by default, so disable it
+    enabled: false
+    configurationFile: /some/other/config.yaml
+```
+
+Note, the enabling/disabling mechanism described in this proposal only works
+when using a structured configuration setup for the apiserver. The
+`--admission-plugin` flag described above does not provide a means to enable or
+disable a plugin (TODO: should it????).
+
+## Code changes
+
+### Admission interfaces
+
+The current interface in `pkg/admission/interfaces.go` looks like this:
+
+```go
+type Interface interface {
+  Admit(a Attributes) (err error)
+  Handles(operation Operation) bool
+}
+```
+
+We'd like to break this into multiple interfaces, one per phase, called at the appropriate times
+within the request lifecycle.
```

Comment:
You'd still have admission plugins, but instead of a plugin having only a single interface to implement, there would be multiple optional interfaces it could implement.

As far as config file -> thing that supports the interfaces, we would register admission plugins just like we do today, by name, and the admission loading code would iterate through all registered admission plugins, look for an appropriately-named file for each plugin, and go from there. Does that answer your question?